CoreScriptsRoot/CoreScripts/PurchasePromptScript2.lua

```diff
@@ -1549,3 +1763,61 @@ if IsNativePurchasing then
 		nativePurchaseFinished(wasPurchased)
 	end)
 end
+
+local function switchGui(activeGui, newVRMode)
```

Comment:
Ever though we try to reset the state when we switch it seems a bit sketchy to me that now GUI points to a different table. Especially when their are spawned loops in the code.
Is there something stopping us from just having two different GUIs in memory and controlling which one we display to the user by the VREnabled prop?